src/main/java/logger/MessageLogger.java

```diff
@@ -0,0 +1,13 @@
+package logger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.Objects;
+
+public class MessageLogger {
+    private static final Logger logger = LoggerFactory.getLogger(MessageLogger.class);
```

Comment:
i'm wondering why do you declare it as static? this obviously complicates testing. make it an instance variable.

Reply:
i read the documentation of slf4j and they used to recomend to declare a logger for each instance of a class but they do not recommend it now. i will leave it like it is now because i do not have a di container now. when i have one i will register it as a service and inject it over the constructor. that way i could test it better.

http://www.slf4j.org/faq.html#declared_static

Reply:
i do not see why do you follow a recommendation that complicates testing.

Reply:
simply followed their recommendations. did not think about testing. i will change this and make it so that it can be tested better.